crypto/pem/pem_lib.c

```diff
@@ -535,9 +536,8 @@ int PEM_get_EVP_CIPHER_INFO(char *header, EVP_CIPHER_INFO *cipher)
         header++;
     }
     *header = '\0';
-    cipher->cipher = enc = EVP_get_cipherbyname(p);
-    *header = c;
-    header++;
+    cipher->cipher = enc = EVP_get_cipherbyname(dekinfostart);
+    *header = separator;
```

Comment:
@richsalz Yes, I know. The problem is having `c` in outer scope is kind inconvenient. And just renaming `c` into `separator` would require making the loop internals much, much longer. The only solution I could come up with is using two distinct variables - one in the outer scope to store the result and another inside the loop.